Index: telepathy-qt-0.9.6.1/tests/dbus/contacts-capabilities.cpp
===================================================================
--- telepathy-qt-0.9.6.1.orig/tests/dbus/contacts-capabilities.cpp
+++ telepathy-qt-0.9.6.1/tests/dbus/contacts-capabilities.cpp
@@ -111,7 +111,7 @@ void TestContactsCapabilities::testCapab
     QStringList ids = QStringList() << QLatin1String("alice")
         << QLatin1String("bob") << QLatin1String("chris");
 
-    gboolean supportTextChat[] = { TRUE, FALSE, FALSE };
+    bool supportTextChat[] = { true, false, false };
 
     TpHandleRepoIface *serviceRepo =
         tp_base_connection_get_handles(TP_BASE_CONNECTION(mConn->service()),
